test/mariner2_test.go

```diff
@@ -304,6 +306,70 @@ echo "$BAR" > bar.txt
 			return gwclient.NewResult(), nil
 		})
 	})
+
+	runTest := func(t *testing.T, f gwclient.BuildFunc) {
+		t.Helper()
+		ctx := startTestSpan(baseCtx, t)
+		testEnv.RunTest(ctx, t, f)
+	}
+
+	t.Run("test mariner2 signing", func(t *testing.T) {
+		t.Parallel()
+		runTest(t, func(ctx context.Context, gwc gwclient.Client) (*gwclient.Result, error) {
+			spec := dalec.Spec{
+				Name:        "foo",
+				Version:     "v0.0.1",
+				Description: "foo bar baz",
+				Website:     "https://foo.bar.baz",
+				Revision:    "1",
+				PackageConfig: &dalec.PackageConfig{
+					Signer: &dalec.Frontend{
+						Image: phonySignerRef,
+					},
+				},
+				Sources: map[string]dalec.Source{
+					"foo": {
+						Inline: &dalec.SourceInline{
+							File: &dalec.SourceInlineFile{
+								Contents: "#!/usr/bin/env bash\necho \"hello, world!\"\n",
+							},
+						},
+					},
+				},
+				Build: dalec.ArtifactBuild{
+					Steps: []dalec.BuildStep{
+						{
+							Command: "/bin/true",
+						},
+					},
+				},
+				Artifacts: dalec.Artifacts{
+					Binaries: map[string]dalec.ArtifactConfig{
+						"foo": {},
+					},
+				},
+			}
+
+			sr := newSolveRequest(withSpec(ctx, t, &spec), withBuildTarget("mariner2/rpm"))
```

Comment:
This is specific to mariner, but we should treat this as a more generic test so that we can add other distros (azlinux3, jammy, etc) and get these tests for free.
Maybe we need to add an extra parameter to the test function to specify the signing target.